nearcache/config_test.go

```diff
@@ -0,0 +1,305 @@
+	config, _ := configWithNearCacheNames(
+		"hazelcast.*",
+		"hazelcast.test.*",
+		"hazelcast.test.sub.*",
+	)
+	assertFalseGetNearCacheConfig(t, config, "com.hazelcast.myNearCache")
+	assertFalseGetNearCacheConfig(t, config, "com.hazelcast.test.myNearCache")
+	assertFalseGetNearCacheConfig(t, config, "com.hazelcast.test.sub.myNearCache")
+}
+
+func TestNearCacheConfigWildcardMatchingPointEndsWith(t *testing.T) {
+	// ported from: com.hazelcast.client.config.MatchingPointConfigPatternMatcherTest#testNearCacheConfigWildcardMatchingPointEndsWith
+	config, _ := configWithNearCacheNames(
+		"*.sub",
+		"*.test.sub",
+		"*.hazelcast.test.sub",
+	)
+	assertFalseGetNearCacheConfig(t, config, "com.hazelFast.Fast.sub.myNearCache")
+	assertFalseGetNearCacheConfig(t, config, "hazelFast.test.sub.myNearCache")
+	assertFalseGetNearCacheConfig(t, config, "test.sub.myNearCache")
+}
+
+func TestNearCacheConfigWildcardOnly(t *testing.T) {
+	// ported from: com.hazelcast.client.config.MatchingPointConfigPatternMatcherTest#testNearCacheConfigWildcardOnly
+	config, ncs := configWithNearCacheNames("*")
+	assert.Equal(t, ncs[0], assertTrueGetNearCacheConfig(t, config, "com.hazelcast.myNearCache"))
+}
+
+func TestNearCacheConfigWildcardOnlyMultipleConfigs(t *testing.T) {
+	// com.hazelcast.client.config.MatchingPointConfigPatternMatcherTest#testNearCacheConfigWildcardOnlyMultipleConfigs
+	config, ncs := configWithNearCacheNames("*", "com.hazelcast.*")
+	assert.Equal(t, ncs[1], assertTrueGetNearCacheConfig(t, config, "com.hazelcast.myNearCache"))
+}
+
+func TestDefaultConfig(t *testing.T) {
+	ncc := nearcache.Config{}
+	if err := ncc.Validate(); err != nil {
+		t.Fatal(err)
+	}
+	target := nearcache.Config{
+		Name:           "default",
+		EvictionConfig: nearcache.EvictionConfig{},
+		PreloaderConfig: nearcache.PreloaderConfig{
+			Enabled:                  false,
+			Directory:                "",
+			StoreIntervalSeconds:     600,
+			StoreInitialDelaySeconds: 600,
+		},
+		InMemoryFormat:    nearcache.InMemoryFormatBinary,
+		SerializeKeys:     false,
+		TimeToLiveSeconds: math.MaxInt32,
+		MaxIdleSeconds:    math.MaxInt32,
+	}
+	assert.Equal(t, target, ncc)
+}
+
+type testCase struct {
+	name string
+	cfg  nearcache.Config
+}
+
+func (tc testCase) Run(t *testing.T) {
+	t.Run(tc.name, func(t *testing.T) {
+		err := tc.cfg.Validate()
+		if !errors.Is(err, hzerrors.ErrInvalidConfiguration) {
+			t.Fatalf("%s: expected ErrInvalidConfiguration", tc.name)
+		}
+	})
+}
+
+func TestConfigInvalid(t *testing.T) {
+	testCases := []testCase{
+		{
+			name: "negative time to live",
+			cfg:  nearcache.Config{TimeToLiveSeconds: -1},
+		},
+		{
+			name: "negative max idle",
+			cfg:  nearcache.Config{MaxIdleSeconds: -1},
+		},
+	}
+	for _, tc := range testCases {
+		tc.Run(t)
+	}
+}
+
+type comparator struct{}
+
+func (c comparator) Compare(a, b nearcache.EvictableEntryView) int {
+	return 0
+}
+
+func TestEvictionConfigInvalid(t *testing.T) {
```

Comment:
We can port validation tests from Java as well. See `testMaxSize_whenValueIsPositive_thenSetValue` and `testMaxSize_whenValueIsNegative_thenThrowException`.

Reply:
Negative size is already tested. I've added more validation tests for others at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/112f3ac833c9d97eb9261bb3b436431ac8177081